### PR TITLE
Remove IISIntegration submodule, add CI [2.2]

### PR DESCRIPTION
**Pull request by jkotalik**

### CODE REVIEW

**Comment by natemcmaster** [comment]
Should we do this in release/2.1 first?

File: .azure/pipelines/jobs/iisintegration_job.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+parameters:
+  agentOs: 'Windows'
+  configuration: 'Release'
+  buildArgs: ''
+  jobName: ''
+
+jobs:
+- template: .vsts-pipelines/templates/project-ci.yml@buildtools
```
Comment (natemcmaster) [inline .azure/pipelines/jobs/iisintegration_job.yml]: I don't think you can mix phases and jobs syntax. You'll have to make this change anywhere "vsts-pipelines" templates are used.

```suggestion
- template: .azure/pipelines/templates/project-ci.yml@buildtools
```

**Comment by jkotalik** [comment]
This PR is mostly testing CI stuff, which is applicable to 2.1. For 2.1, we can probably just add a folder name here: https://github.com/aspnet/AspNetCore/pull/3865/files#diff-4be1923f2886cfee6a9be73685daa857R25. 

I'll make a PR shortly to remove the submodule in 2.1 and merge it up.

File: .azure/pipelines/fast-pr-validation.yml [hunk 2]
```diff
@@ -18,12 +18,61 @@ phases:
...
     matrix:
       DataProtection:
         _FolderName: DataProtection
       WebSockets:
         _FolderName: WebSockets
+
+- template: jobs/iisintegration_job.yml
+  parameters:
+    jobName: IISIntegration_IIS_Windows
+    buildArgs: "/p:SkipIISBackwardsCompatibilityTests=true /p:SkipIISExpressTests=true /p:SkipIISForwardsCompatibilityTests=true"
```
Comment (natemcmaster) [inline .azure/pipelines/fast-pr-validation.yml]: Looks like there is a lot of redundancy. Another way to do this is with a `matrix`. https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=vsts&tabs=schema#matrix

Something like this ... (didn't fully flesh out all combinations, but you get the idea):

```yml

- template: jobs/iisintegration_job.yml
  parameters:
    variables:
      _FolderName: IISIntegration
    matrix:
      IISIntegration_IIS:
        TestGroupName: IIS
        SkipIISBackwardsCompatibilityTests: true
        SkipIISExpressTests: true
        SkipIISForwardsCompatibilityTests: true
      IISIntegration_IISExpress:
        TestGroupName: IISExpress
        SkipIISBackwardsCompatibilityTests: true
        SkipIISTests: true
        SkipIISForwardsCompatibilityTests: true
      IISIntegration_IISForwardCompat:
        TestGroupName: IISForwardCompat
        SkipIISBackwardsCompatibilityTests: true
        SkipIISTests: true
        SkipIISForwardsCompatibilityTests: false
    jobName: IISIntegration_$(TestGroupName)
    buildArgs: "/p:SkipIISBackwardsCompatibilityTests=$(SkipIISBackwardsCompatibilityTests) /p:SkipTests=$(SkipIISTests)  /p:SkipIISExpressTests=$(SkipIISExpressTests) /p:SkipIISForwardsCompatibilityTests=$(SkipIISForwardsCompatibilityTests)"

```

Comment (jkotalik) [reply .azure/pipelines/fast-pr-validation.yml]: Yeah this is much better.

**Comment by natemcmaster** [comment]
Talked about this in person. Let's merge the submodule removal without the Azure Pipelines YML changes for now to unblock this: https://github.com/aspnet/AspNetCore-Internal/issues/1264

**Review by natemcmaster — APPROVED** [review]

**Comment by jkotalik** [comment]
@natemcmaster 🆙 📅 (if you actually get this message).

**Pull request merged**